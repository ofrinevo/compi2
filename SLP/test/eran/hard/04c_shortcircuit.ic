
class A {
	int field ;  
	
	int incField () { 
		field = field + 1 ; 
		return field ; 
	}
	int print() { 
		Library.println("This line will not be printed") ; 
		return 1 ; 
	}
	
	static void main (string[] args) { 
		A a = new A() ; 
		a.field = 9 ; 
		
		int x = 4 ; 
		int res = 0 ; 
		
		if ( (x > 3) || (x/0 > 0) ) 
			res = res + 1 ; 
		if ( (x < 3) && ( a.incField() > 10 ) ) 
			res = res + 10 ; 
		if ( (x == 4) || ( a.print() == 0 ) )
			res = res + 100 ; 
		
		Library.print("res = ") ; 
		Library.printi(res) ; 
		Library.println("") ; 
		Library.print("a.field = ") ; 
		Library.printi(a.field) ; 
		
	}

}
