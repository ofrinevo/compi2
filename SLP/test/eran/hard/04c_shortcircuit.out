res = 101
a.field = 9
